tests/test_association_task.py

```diff
@@ -95,64 +100,56 @@ def tearDown(self):
         del self.exposure
 
     def test_run(self):
-        """ Test the run method with a database that already exists and
+        """Test the run method with a database that already exists and
         contains DIAObjects and Sources.
         """
-        dia_collection = self._run_association_and_retrieve_objects(True)
+        dia_objects = self._run_association_and_retrieve_objects(True)
         not_updated_idx = 0
         updated_idx_start = 1
         new_idx_start = 5
         total_expected_dia_objects = 10
-        self.assertEqual(len(dia_collection.dia_objects),
-                         total_expected_dia_objects)
-        for obj_idx, output_dia_object in \
-                enumerate(dia_collection.dia_objects):
+        self.assertEqual(len(dia_objects), total_expected_dia_objects)
+        for obj_idx, dia_object in enumerate(dia_objects):
             if obj_idx == not_updated_idx:
                 # Test the DIAObject we expect to not be associated with any
                 # new DIASources.
-                self.assertEqual(output_dia_object.n_dia_sources, 2)
+                self.assertEqual(dia_object['n_dia_sources'], 1)
+                self.assertEqual(dia_object.getId(), obj_idx)
             elif updated_idx_start <= obj_idx < new_idx_start:
                 # Test that associating to the existing DIAObjects went
                 # as planned and test that the IDs of the newly associated
                 # DIASources is correct.
-                self.assertEqual(output_dia_object.n_dia_sources, 3)
-                self.assertEqual(
-                    output_dia_object.dia_source_catalog[-1].getId(),
-                    obj_idx - 1 + 10)
+                self.assertEqual(dia_object['n_dia_sources'], 2)
```

Comment:
Why did the number of sources change?

Reply:
Just because the number of sources I store in the DB changed. It still tests the same actions of successfully updating the values in the DIAObject table and associating/appending to known DIAObjects.